src/netlink.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2019 Xaptum, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "netlink.h"
+
+#include <net/if.h>
+#include <fcntl.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "log.h"
+#include "conn_state.h"
+
+static
+int
+udp_check(struct enftun_netlink* nl)
```

Comment:
Move this into the `conn_state` class.

Keep the netlink class narrowly focused on netlink.  The eventual ping class narrowly focused on pings. And then conn_state class has the logic to tie them all together --- when netlink indicates a state change, do the udp address check, wait for a ping response, etc.